source/game.programme.newsevent.bmx

```diff
@@ -293,86 +323,94 @@ Type TNewsEventCollection
 
 	Method GetNewsHistory:TNewsEvent[](limit:Int=-1)
 		If limit = -1
-			Return newsEventsHistory[.. newsEventsHistoryIndex]
+			Return newsEventsHistory
 		Else
 			Return newsEventsHistory[Max(0,newsEventsHistoryIndex-limit) .. newsEventsHistoryIndex]
 		EndIf
 	End Method
 
 
 	'returns (and creates if needed) a list containing only follow up news
-	Method GetFollowingNewsList:TList(genre:Int=-1)
+	Method GetFollowingNewsList:TObjectList(genre:Int=-1)
 		'create if missing
 		If Not _followingNewsEvents Then _InvalidateFollowingNewsEvents()
 
-		If Not _followingNewsEvents[genre+1]
-			_followingNewsEvents[genre+1] = CreateList()
+		If Not _followingNewsEventsValid[genre+1]
+			If Not _followingNewsEvents[genre+1]
+				_followingNewsEvents[genre+1] = New TObjectList()
+			EndIf
 			For Local event:TNewsEvent = EachIn newsEvents.Values()
 				If event.newsType <> TVTNewsType.FollowingNews Then Continue
 				'only interested in a specific genre?
 				If genre <> -1 And event.GetGenre() <> genre Then Continue
 
 				_followingNewsEvents[genre+1].AddLast(event)
 			Next
+			_followingNewsEventsValid[genre+1] = True
 		EndIf
 		Return _followingNewsEvents[genre+1]
 	End Method
 
 
 	'returns (and creates if needed) a list containing only initial news
-	Method GetUpcomingNewsList:TList(genre:Int=-1)
+	Method GetUpcomingNewsList:TObjectList(genre:Int=-1)
 		'create if missing
 		If Not _upcomingNewsEvents Then _InvalidateUpcomingNewsEvents()
 
-		If Not _upcomingNewsEvents[genre+1]
-			_upcomingNewsEvents[genre+1] = CreateList()
+		If Not _upcomingNewsEventsValid[genre+1]
+			If Not _upcomingNewsEvents[genre+1]
+				_upcomingNewsEvents[genre+1] = New TObjectList()
+			EndIf
 			For Local event:TNewsEvent = EachIn newsEvents.Values()
-				'skip events already happened or not happened at all (-> "-1")
-				If event.HasHappened() Or event.happenedTime = -1 Then Continue
+				'skip events already happened (and processed) or not
+				'happened at all (-> "-1")
+				If event.HasFlag(TVTNewsFlag.HAPPENING_PROCESSED) Or event.happenedTime = -1 Then Continue
 				'only interested in a specific genre?
 				If genre <> -1 And event.GetGenre() <> genre Then Continue
 
 				_upcomingNewsEvents[genre+1].AddLast(event)
 			Next
+			_upcomingNewsEventsValid[genre+1] = True
 		EndIf
 		Return _upcomingNewsEvents[genre+1]
 	End Method
+	
+	
+	'protect a thread id until the last news of that thread has happened
+	Method ProtectNewsThread:Int(news:TNewsEvent)
```

Comment:
OK. Bei mehrfachem Drüberschauen war mir entgangen, dass das protect thread in eine eigene Methode gewandert ist.